Backend/app/routes/notification.py

```
@@ -0,0 +1,210 @@
+                raise HTTPException(status_code=401, detail="Invalid token payload: no user id (HS256)")
+            return {"id": user_id}
+        else:
+            logger.warning("No HS256 secret set, skipping HS256 check.")
+    except Exception as e:
+        logger.error(f"HS256 verification failed: {str(e)}")
+    logger.error("Both RS256 and HS256 verification failed.")
+    raise HTTPException(status_code=401, detail="Invalid token: could not verify with RS256 or HS256.")
+
+@router.get("/", response_model=List[dict])
+async def get_notifications(
+    db: AsyncSession = Depends(get_db),
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)